### PR TITLE
Lep 432 employment translator

### PR DESCRIPTION
**Pull request by starswan**
## What does this pull request do?

Translator for employment figures

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: cla_backend/apps/reports/tests/test_translate_savings.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from unittest import TestCase
```
Comment (davidread) [inline cla_backend/apps/reports/tests/test_translate_savings.py:1]: The Django convention appears to be `from django.test import TestCase`

File: cla_backend/libs/eligibility_calculator/calculator.py [hunk 2]
```diff
@@ -331,17 +335,23 @@ def is_disposable_capital_eligible(self):
         return self.disposable_capital_assets <= limit
 
     def is_eligible(self):
-        cfe_request_dict = self._translate_case()
-
         # this line fails the lint check as we're ignoring the response
```
Comment (davidread) [inline cla_backend/libs/eligibility_calculator/calculator.py:338]: this comment is a bit distracting - best to use suffix the line with `# noqa: F401` (or whatever the code is for the lint check to ignore)

File: cla_backend/libs/eligibility_calculator/cfe_civil/__init__.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# translate bank_balance, investment_balance and asset_balance from CLA into CFE format
+# credit_balance isn't captured (and sounds like a credit card balance) which is not part of eligibility calculations
```
Comment (davidread) [inline cla_backend/libs/eligibility_calculator/cfe_civil/__init__.py:2]: We shouldn't assume stuff like this - let's do the analysis to get a definite answer

File: cla_backend/libs/eligibility_calculator/cfe_civil/__init__.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# translate bank_balance, investment_balance and asset_balance from CLA into CFE format
```
Comment (davidread) [inline cla_backend/libs/eligibility_calculator/cfe_civil/__init__.py:1]: Shouldn't have much logic in a __init__.py file. This function is a duplicate of savings.py, so I think you just forgot to delete this copy

File: cla_backend/libs/eligibility_calculator/cfe_civil/__init__.py (hunk 3 above)
Comment (davidread) [reply cla_backend/libs/eligibility_calculator/cfe_civil/__init__.py:2]: but this is not to do with employment anyway, so can leave it til later

File: cla_backend/libs/eligibility_calculator/cfe_civil/income.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+                "benefits_in_kind": 0,
+                "tax": deductions.income_tax,
+                "national_insurance": deductions.national_insurance,
+                "prisoner_levy": 0,
+                "student_debt_repayment": 0
+            }
+    }
+    if income.self_employed:
+        return {
+            "self_employment_details": [ value ]
```
Comment (davidread) [inline cla_backend/libs/eligibility_calculator/cfe_civil/income.py:16]: "receiving_only_statutory_sick_or_maternity_pay" is for employment-only, so will get rejected by CFE, I believe
Screenshot 2023-12-05 at 11 34 53

**Comment by davidread** [comment]
Do we handle the £45 employment allowance acceptably? There are separate options employed and self-employed:
Screenshot 2023-12-05 at 11 40 22
But when the data gets to us it seems to be one or the other

File: cla_backend/libs/eligibility_calculator/cfe_civil/income.py (hunk 5 above)
Comment (starswan) [reply cla_backend/libs/eligibility_calculator/cfe_civil/income.py:16]: have added a test that proved it - had to remove benefits_in_kind as well

File: cla_backend/apps/reports/tests/test_translate_savings.py (hunk 1 above)
Comment (starswan) [reply cla_backend/apps/reports/tests/test_translate_savings.py]: that seems to bring in a database. There seem to be many simple examples that use this style when not dependant on Django

File: cla_backend/libs/eligibility_calculator/cfe_civil/__init__.py (hunk 3 above)
Comment (starswan) [reply cla_backend/libs/eligibility_calculator/cfe_civil/__init__.py]: I've un-tangled the employment and savings PRs as they didn't really benefit from being together

**Comment by davidread** [comment]
just check the £45 again pls

**Review by ripan — APPROVED** [review]

File: cla_backend/libs/eligibility_calculator/calculator.py [hunk 6]
```diff
@@ -340,15 +344,14 @@ def _do_cfe_civil_check(self):
...
+        return CfeResponse(cfe_civil_response.content)
 
     def _translate_case(self):
+        submission_date = datetime.date(2022, 5, 19)
         # produce the simplest possible plain request to CFE to prove the route
-        default_request_data = {
+        request_data = {
             "assessment": {
-                "submission_date": "2022-05-19",
+                "submission_date": str(submission_date),
```
[the remark this answers is not in the document]
Comment (starswan) [reply cla_backend/libs/eligibility_calculator/calculator.py:354]: we are passing submission date to translators

**Pull request merged**